Separate rpmdb data to rpmdb table, left join via %rpm% field name on demand.

As the rpmdb data is huge comparing to other data, it's best we keep it
in separate table, if it is still a performance problem, we could use
separate datafile.

You need to create 'rpm' table manually:
CREATE TABLE rpmdb (eid INTEGER PRIMARY KEY, pid INTEGER, value);

Handling it automatically would mean have to renumber upstream patches each
time...

#--- dokuwiki-plugin-data-20120227/db/latest.version~	2012-02-27 16:32:51.000000000 +0200
#+++ dokuwiki-plugin-data-20120227/db/latest.version	2012-03-13 13:10:57.241251292 +0200
#@@ -1 +1 @@
#-5
#+6
--- dokuwiki-plugin-data-20120227/helper.php~	2012-03-13 13:10:04.000000000 +0200
+++ dokuwiki-plugin-data-20120227/helper.php	2012-03-13 13:11:33.742793083 +0200
@@ -184,6 +184,7 @@
         static $specials = array('%title%'   => array('page', 'title'),
                                  '%pageid%'  => array('title', 'page'),
                                  '%class%'   => array('class'),
+                                 '%rpm%'     => array('rpm'),
                                  '%lastmod%' => array('lastmod','timestamp'));
         if (isset($specials[$column['title']])) {
             $s = $specials[$column['title']];
--- dokuwiki-plugin-data-20120227/lang/en/lang.php~	2012-02-27 16:32:51.000000000 +0200
+++ dokuwiki-plugin-data-20120227/lang/en/lang.php	2012-03-13 13:12:04.190745746 +0200
@@ -16,6 +16,7 @@
 $lang['page']      = 'Page';
 $lang['title']     = 'Page Name';
 $lang['class']     = 'Page Class';
+$lang['rpm']       = 'RPM Package';
 $lang['lastmod']   = 'Last Modified';
 
 $lang['name']      = 'Type Alias';
--- dokuwiki-plugin-data-20100608/syntax/table.php	2010-11-03 15:20:30.000000000 +0200
+++ dokuwiki-plugin-data-20100608/syntax/table.php	2010-12-01 19:50:59.217678000 +0200
@@ -328,6 +328,13 @@
                 $select[] = '" " || pages.class';
             }elseif($key == '%title%'){
                 $select[] = "pages.page || '|' || pages.title";
+            }elseif($key == '%rpm%'){
+                if(!isset($tables[$key])){
+                    $tables[$key] = 'T'.(++$cnt);
+                    $from  .= ' LEFT JOIN rpmdb AS '.$tables[$key].' ON '.$tables[$key].'.pid = pages.pid';
+                }
+                $select[] = "group_concat(".$tables[$key].".value,'\n')";
+
             }else{
                 if(!isset($tables[$key])){
                     $tables[$key] = 'T'.(++$cnt);
